### ISSUE TITLE
(Non?) Issue with multiple keys

### ISSUE THREAD

**Issue by TomMD** [tsani/servant-github-webhook#4]
It seems painful that the number of keys an application can accept is static.  In my view most uses for webhooks are services which can be offered to an ever-widening set of users and repositories.  If that's the case then either this library isn't intended for this use or I misunderstand the guarantees github provides about the secrecy of a key - it is not accessible to the repository owner? Can one key be re-used for many users and many repositories?

**Comment by tsani** [comment]
The use-case I had for this library when I wrote it was a special-purpose application that received webhooks only from a few repositories that I controlled, so the fact that the number of keys was static wasn't really a problem for me.

The library isn't currently intended for situations where the set of keys is dynamic, but it wouldn't be too difficult to get dynamic keys. Currently, users have to write a function that computes the hashing key given the (statically fixed) index of the repository. We could extend this so that the user function obtains also some information obtained from the webhook, such as the repository name. The user function could then use this information to look up the key in a database.

Regarding key secrecy, github keeps keys very secret. Even the repository owner can't see what the key is once they set it. However, nothing is stopping someone from reusing the same key in multiple repositories.
